lib/iris/_lazy_data.py

```diff
@@ -386,8 +387,20 @@ def map_complete_blocks(src, func, dims, out_sizes):
     data = data.rechunk(in_chunks)
 
     # Determine output chunks
+    drop_axis = []
     out_chunks = list(data.chunks)
     for dim, size in zip(dims, out_sizes):
-        out_chunks[dim] = size
-
-    return data.map_blocks(func, chunks=out_chunks, dtype=src.dtype)
+        if size == 0:
+            drop_axis.append(dim)
+        else:
+            out_chunks[dim] = size
+
+    out_chunks = [
```

Comment:
:+1:
I was sure I could code this more clearly, but everything I tried make it worse !